lib/active_model/serializer/associations.rb

```diff
@@ -82,14 +82,17 @@ def associate(reflection)
         end
       end
 
+      # @param [IncludeTree] include_tree (optional)
       # @return [Enumerator<Association>]
       #
-      def associations
+      def associations(include_tree = nil)
         return unless object
 
+        include_tree ||= ActiveModel::Serializer::IncludeTree.from_string('*')
+
         Enumerator.new do |y|
           self.class._reflections.each do |reflection|
-            y.yield reflection.build_association(self, instance_options)
+            y.yield reflection.build_association(self, instance_options) if include_tree.key?(reflection.name)
```

Comment:
Having a conditional at the end of a long line is hard to read (i.e. easy to miss).  Better:

```ruby
each do |reflection|
  next unless include_tree.key?(reflection.name)
  association = reflection.build_association(self, instance_options)
  y.yield association
...
end
```